sections/3-architecture.md

```diff
@@ -156,4 +156,53 @@ if you don't need memoization.
 
 Also, make sure to read the
 [Rule 3.3](#33-should-not-apply-optimizations-without-profiling) on
-profiling.
\ No newline at end of file
+profiling.
+
+### 3.5. MUST NOT use parameterless ConfigFactory.load() or access a Config object directly
+
+It may be very tempting to call the oh-so-available-and-parameterless `ConfigFactory.load()` method whenever you need to pull something from the configuration, but doing so WILL boomerang back at you, for instance when writing tests. 
+
+If you have [`ConfigFactory.load()`](https://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigFactory.html#load--) scattered all around your classes they are basically loading the default configuration when your code runs, which more often than not, is not what you really want to happen in a testing environment, where you need to have a modified configuration loaded (e.g., different timeouts, different implementations, different IPs, etc.).
+
+One way to go about dealing with it, is to pass the `Config` instance itself to whomever needs it, or have the needed values from it passed in.
+
+The situation described here is in fact a flavour of the [prefer dependency injection (DI) over Service Locator](http://stackoverflow.com/questions/1638919/how-to-explain-dependency-injection-to-a-5-year-old/1638961#1638961) practise.
+
+One of the (very) few exceptions to avoiding `ConfigFactory.load()`, is if it's called once in your application's root, say in your `main()` (or equivalent) so that you don't have to hardcode your configuration's filename.
+
+Another common practise it to have domain specific config classes, which are parsed from the general purpose, map-like, Config objects. The benefit of this approach is that specialized config classes faithfully represent your specific configuration needs, and once parsed, allow you to work against compiled classes in a safer manner (where "safer" means you do config.ip, instead of config.getString("ip")).
+
+This also has the benefit of clarity, as your domain specific config class conveys the needed properties in a more explicit and readable manner.
+
+It should be noted that this approach comes on the account of your config object's flexibly, since you will need to change its code upon introducing new configuration properties, but since you will need to introduce new code in order to utilize the new property anyway, the overhead might be very worthwhile.
+
+Consider the following example:
+
+```scala
+  /*
+  this is your domain specific config class, with a pre-defined set of
+  properties you've modelled according to your domain, as opposed to
+  a map-like properties bag
+   */
+  case class MyComponentConfig(timeout: Long, ip: String)
+
+  // this parse method parses the map-like properties bag into
+  // a domain config class
+  def parseConfigProperties(properties: Config): MyComponentConfig = {
+    MyComponentConfig(properties.getLong("application.timeout"),
```

Comment:
These case classes can easily grow, which is why I prefer named arguments instead of relying on positioning, like in my sample. Since these properties are primitives, it's easy to mix them and named arguments makes it more resistant to change, even if redundant.